coffee_github/src/utils.rs

```diff
@@ -48,3 +48,27 @@ pub async fn git_upgrade(
         Ok(UpgradeStatus::Updated(upstream_commit, date))
     }
 }
+
+pub async fn git_upgrade_with_git_head(
+    path: &str,
+    git_head: &str,
+    verbose: bool,
+) -> Result<UpgradeStatus, CoffeeError> {
+    use tokio::process::Command;
```

Comment:
idk if this is a good idea, the checkout is just `git checkout <branch-name>` 

Why we need to use all this code?